physical/raft/fsm.go

```diff
@@ -36,6 +36,10 @@ const (
 	putOp
 	restoreCallbackOp
 	getOp
+	verifyReadOp
+	verifyListOp
+	beginTxOp
+	commitTxOp
```

Comment:
`beginTxOp` / `commitTxOp` are mostly transaction sentinels for the time being. However, when implementing the single hash method (number 3), we can send the hash type in `beginTxOp` and the final hash in `commitTxOp`. We'll need `verifyReadOp` because while we can re-order any un-modified reads to occur at the end (sending them along in the `commitTxOp` if we wanted fewer operations in the log), we can't avoid sending reads before writes. We'll need `verifyListOp`, because its results can be impacted by writes and we won't necessarily know to compact (or not compact) them, so we'd prefer to keep them in execution order.